subprojects/diagnostics/src/integTest/groovy/org/gradle/api/plugins/ProjectReportsPluginIntegrationTest.java

```diff
@@ -16,18 +16,85 @@
 package org.gradle.api.plugins;
 
 import org.gradle.integtests.fixtures.AbstractIntegrationTest;
+import org.gradle.integtests.fixtures.executer.ExecutionResult;
+import org.gradle.util.TextUtil;
 import org.junit.Test;
 
 public class ProjectReportsPluginIntegrationTest extends AbstractIntegrationTest {
     @Test
     public void generatesReportFilesToReportsDirectory() {
-        testFile("build.gradle").writelns(
-                "apply plugin: 'project-report'"
-        );
+        applyProjectReportPlugin();
         inTestDirectory().withTasks("projectReport").run();
 
         testFile("build/reports/project/dependencies.txt").assertExists();
         testFile("build/reports/project/properties.txt").assertExists();
         testFile("build/reports/project/tasks.txt").assertExists();
     }
+
+    private void applyProjectReportPlugin() {
+        testFile("build.gradle").writelns(
+            "apply plugin: 'project-report'"
+        );
+    }
+
+    @Test
```

Comment:
We prefer Spock tests, but we have a mix of older test classes (like this one) that don't use it.  We usually convert tests as we find them, but I don't want to ask you to do that. I went ahead and converted it: https://github.com/gradle/gradle/blob/master/subprojects/diagnostics/src/integTest/groovy/org/gradle/api/plugins/ProjectReportsPluginIntegrationSpec.groovy

If you wouldn't mind deleting this class and putting your tests into `ProjectReportsPluginIntegrationSpec` that would be appreciated.  Having a parameterized test sounds fine.  I don't think we have to assert on the exact message (with the full path).  Just asserting that the output contains "See the report at:" is probably enough (that's what we do elsewhere).